### DISCUSSION TITLE
Consider changing Literal None fields

### DISCUSSION THREAD

**Discussion by jtyoung84** [AllenNeuralDynamics/aind-data-schema#822] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently, fields which are constant and None are described as:
```python
abbreviation: Literal[None] = Field(None)
```
However, the type goes missing when rendering the schema:
```
"abbreviation": {
      "const": null,
      "default": null,
      "title": "Name"
    }
```

**Describe the solution you'd like**
We can get the same behavior by just making the field a NoneType:
```python
abbreviation: type(None) = Field(None)
```
which renders as:
```
"abbreviation": {
      "default": null,
      "title": "Bar",
      "type": "null"
    }
```

**Describe alternatives you've considered**
We can leave things as they are, but we'd have to modify the schema when rendering to guis.

**Additional context**

**Comment by bruno-f-cruz** [comment]
The problem is not so much the null type as it is the Literal annotation, I think we should fix this at the level of the model creation instead. Currently, if you make a Literal of type string, you have no way to know what is the type in the schema either.

Overloading this method should do the trick:
https://github.com/pydantic/pydantic/blob/9811727b1353e605d7183bae3d4e0d1d4ba30b95/pydantic/json_schema.py#L722

I have implemented the solution in another project where I am making my custom json model creator since the default pydantic one has some rough edges.
https://github.com/AllenNeuralDynamics/Aind.Behavior.Services/blob/aafc4da15293e3501106293173bf393bde26309f/src/DataSchemas/aind_behavior_services/utils.py#L44

Happy to make a PR if you want to go in this direction, as it would make my life easier as well.

**Comment by bruno-f-cruz** [comment]
Tracking https://github.com/pydantic/pydantic/issues/8905

**Comment by bruno-f-cruz** [comment]
As the previously mentioned Pydantic issue was fixed, this discussion should be turned into a new issue to update pydantic to the latest release, once available (currently tracked in https://github.com/pydantic/pydantic/issues/8993).

**Reply by bruno-f-cruz** [reply]
Turns out the previous PR was missing the `Null` case. This has now been fixed via https://github.com/pydantic/pydantic/pull/9135 and merged to main. We should keep an eye and bump pydantic once this change is released.
